agent/wpthook/test_state.cc

```diff
@@ -172,6 +173,11 @@ void TestState::Start() {
   GetSystemTime(&_start_time);
   if (!_start.QuadPart)
     _start.QuadPart = _step_start.QuadPart;
+
+  //This is only called once, on the first navigate

```

Comment:
Not sure what you mean by the first navigate, but TestState::Start() is called on every script step that is of record type. Like 'navigate', 'xxxAndWait', 'submitForm', etc. That means, for each such blocking steps, the _timeout_start_time will be reset to the system time at that point.